src/dense.jl

```diff
@@ -186,14 +186,19 @@ Defined by [ADTypes.jl](https://github.com/SciML/ADTypes.jl).
 
 # Constructors
 
-    AutoReverseDiff(; compile=false)
+    AutoReverseDiff(; compile::Union{Val, Bool} = Val(false))
```

Comment:
changing the default value of `compile` from `false` to `Val(false)` is breaking, we should keep `false`

Reply:
No it's not because it also handles the bool.

Reply:
That wouldn't be breaking because the final object being created is what is accessible to the user, and there is no way to distinguish between `AutoReverseDiff(; compile=false)` and `AutoReverseDiff(; compile=Val(false))` (except the case I mention next).

Let's say we keep `compile = false`, then the following fails:

```julia
@inferred AutoReverseDiff()
```

But changing default to `Val` makes it indistinguishable.

Reply:
There is a way to distinguish these objects, which is when you access the compile field and expect a Bool. That is part of the current API and users of ADTypes (like DI) rely on it.

Reply:
it is a bool

Reply:
Right, my bad